### PR TITLE
[development BT-14] Added extra field Username

### PR DESCRIPTION
**Pull request by Prakruthee**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ gem 'uglifier', '>= 1.3.0'
 # Use CoffeeScript for .js.coffee assets and views
 gem 'coffee-rails', '~> 4.0.0'
 # See https://github.com/sstephenson/execjs#readme for more supported runtimes
-#gem 'therubyracer',  platforms: :ruby
```
Comment (kedarnag138) [inline Gemfile:15]: Why was this changed ? Meaning Gemfile.

File: app/assets/javascripts/signup.js [hunk 2]
```diff
@@ -50,7 +67,7 @@ function UserValidator() {
   "user[password_confirmation]": {
     required: "This field is required",
     equalTo:"Passwords do not match!",
-    pswd_match: "Password matches"
+    pswd_match: "Password doesn't matches"
```
Comment (kedarnag138) [inline app/assets/javascripts/signup.js]: Password does not match

File: Gemfile (hunk 1 above)
Comment (Prakruthee) [inline Gemfile:15]: For heroku deployment I had commented the therubyracer gem but we need that gem for IT deployment.

**Comment by Prakruthee** [comment]
I have done the changes and push the code

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -12,28 +12,35 @@
...
+    it { should allow_value('something@domain.com').for(:email)}
+    it { should allow_value('PrakruthiS').for(:username)}
+    it { should allow_value('password@1').for(:password)}
   end
 
-  it "should validate name length" do
+it "should validate username length" do
 
+    user.username = "sg"*256
+    user.valid?
```
Comment (asimha) [inline spec/models/user_spec.rb:28]: Why to check "user.valid?" here, what is it helping ?
